### PR TITLE
[RW-11062][risk=low] Fix currentWorkspaceStore has incorrect (previous workspace)

### PR DESCRIPTION
**Pull request by yonghaoy**
Workspace: https://github.com/all-of-us/workbench/blob/main/ui/src/app/pages/workspace/workspace-wrapper.tsx#L137
is populated by `withCurrentWorkspace`. 
 But `withCurrentWorkspace` returns previous workspace.  

The change is small, most lines are done by yarn lint. Is that expected? 

---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/pages/workspace/workspace-wrapper.tsx [hunk 1]
```diff
@@ -133,156 +132,154 @@ const MultiRegionWorkspaceNotification = () => {
...
-        }
-      };
-    }, []);
-
-    useEffect(() => {
-      if (workspace) {
-        maybeStartPollingForUserApps(workspace.namespace);
+export const WorkspaceWrapper = ({ hideSpinner }) => {
+  const params = useParams<MatchParams>();
+  const { ns, wsid } = params;
```
Comment (yonghaoy) [inline ui/src/app/pages/workspace/workspace-wrapper.tsx:137]: Move this to the top and use ns as south of truth of workspace namespace

File: ui/src/app/pages/workspace/workspace-wrapper.tsx [hunk 2]
```diff
@@ -133,156 +132,154 @@ const MultiRegionWorkspaceNotification = () => {
...
+      if (nextWs && nextWs.namespace === ns && nextWs.id === wsid) {
+        currentWorkspaceStore.next(nextWs);
+        updateStores(ns);
+      } else {
+        getWorkspaceAndUpdateStores(ns, wsid);
       }
-    }, [ns, wsid]);
+    }
+  }, [ns, wsid]);
+  const workspace = currentWorkspaceStore.getValue();
```
Comment (yonghaoy) [inline ui/src/app/pages/workspace/workspace-wrapper.tsx:228]: I added this line here. Previously, workspace is from `withCurrentWorkspace`. I change to `currentWorkspaceStore.getValue()` and I think by this time currentWorkspaceStore already have correct workspace. Is that correct?

File: ui/src/app/pages/workspace/workspace-wrapper.tsx [hunk 3]
```diff
@@ -133,156 +132,154 @@ const MultiRegionWorkspaceNotification = () => {
   );
 };
 
-export const WorkspaceWrapper = fp.flow(withCurrentWorkspace())(
-  ({ workspace, hideSpinner }) => {
```
Comment (yonghaoy) [inline ui/src/app/pages/workspace/workspace-wrapper.tsx:137]: this workspace is wrong. It is the previous workspace from withCurrentWorkspace().

File: ui/src/app/pages/workspace/workspace-wrapper.tsx (hunk 2 above)
Comment (jmthibault79) [reply ui/src/app/pages/workspace/workspace-wrapper.tsx:228]: I don't think we are guaranteed to have the `currentWorkspaceStore` correctly initialized at this point.

What I'd prefer is a state variable, like `const [workspace, setWorkspace] = useState<Workpace>(undefined)` which we can set when we know we have a valid workspace, like inside the `useEffect` somewhere

File: ui/src/app/pages/workspace/workspace-wrapper.tsx [hunk 4]
```diff
@@ -133,156 +134,157 @@ const MultiRegionWorkspaceNotification = () => {
...
-        } catch {
-          // Ignore InitialRuntimeNotFoundError.
-          // Ignore ExceededActionCountError. This is thrown when the runtime doesn't exist, or
-          // isn't started. Both of these scenarios are expected, since we don't want to do any lazy
-          // initialization here.
-          // Also ignore LeoRuntimeInitializationAbortedError - this is expected when navigating
-          // away from a page during a poll.
-          // Ideally, we would handle or log errors except the ones listed above.
+        updateStores(wsResponse.workspace.namespace);
+        setWorkspace(wsResponse.workspace);
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-wrapper.tsx:206]: looks good - I think we also need a call on line 226-227 for the case where we populate it from `nextWs`

**Review by Qi77Qi — APPROVED** [review]
not react expert..but poked around a bit and things are working as expected as I move around workspaces

**Review by jmthibault79 — APPROVED** [review]
works locally.  thanks for catching and fixing this!

**Pull request merged**